src/ImageSharp/Formats/Tiff/TiffDecoderMetadataCreator.cs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using SixLabors.ImageSharp.Metadata;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+using SixLabors.ImageSharp.Metadata.Profiles.Icc;
+using SixLabors.ImageSharp.Metadata.Profiles.Iptc;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Formats.Tiff
+{
+    /// <summary>
+    /// The decoder metadata creator.
+    /// </summary>
+    internal static class TiffDecoderMetadataCreator
+    {
+        public static ImageMetadata Create<TPixel>(List<ImageFrame<TPixel>> frames, List<TiffFrameMetadata> framesMetaData, bool ignoreMetadata, ByteOrder byteOrder)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            DebugGuard.IsTrue(frames.Count == framesMetaData.Count, nameof(frames), "Image frames and frames metadata should be the same size.");
+
+            if (framesMetaData.Count < 1)
+            {
+                TiffThrowHelper.ThrowImageFormatException("Expected at least one frame.");
+            }
+
+            var imageMetaData = new ImageMetadata();
+            TiffFrameMetadata rootFrameMetadata = framesMetaData[0];
+            SetResolution(imageMetaData, rootFrameMetadata);
+
+            TiffMetadata tiffMetadata = imageMetaData.GetTiffMetadata();
+            tiffMetadata.ByteOrder = byteOrder;
+            tiffMetadata.BitsPerPixel = GetBitsPerPixel(rootFrameMetadata);
+
+            if (!ignoreMetadata)
+            {
+                for (int i = 0; i < frames.Count; i++)
+                {
+                    ImageFrame<TPixel> frame = frames[i];
+                    ImageFrameMetadata frameMetaData = frame.Metadata;
+                    if (frameMetaData.XmpProfile == null)
+                    {
+                        IExifValue<byte[]> val = frameMetaData.ExifProfile.GetValue(ExifTag.XMP);
+                        if (val != null)
+                        {
+                            frameMetaData.XmpProfile = val.Value;
+                        }
+                    }
+
+                    if (imageMetaData.IptcProfile == null && TryGetIptc(frameMetaData.ExifProfile.Values, out byte[] iptcBytes))
```

Comment:
Ok, moved to `ImageFrameMetadata` with e2bd192c

Reply:
@JimBobSquarePants @brianpopow About *pure* metadata tags (Copyright, Artist,... XMP, IPTC profiles). 

These tags are matter for the whole file, they should be stored only in one frame (it should be first frame). It seems that I have met files a couple of times where they were not stored in the first frame, so, just in case, I foreach all frames and first non null take as image profile. 

I couldn't find any documentation about this, I know only from practice. This can be checked - add/edit metadata using `Adobe Bridge` or `XnView` and view tags through `AWare Systems TIFF Tag Viewer`. You may also notice that programs like these don't edit metadata for individual frames, only for image as whole.

So, profiles can be store in image metadata. Also they can be store in frame metadata and user can access from first frame metadata.

Reply:
> I couldn't find any documentation about this, I know only from practice. This can be checked - add/edit metadata using for example `Adobe Bridge` or `XnView` and view tags through `AWare Systems TIFF Tag Viewer`. You may also notice that programs like these don't edit metadata for individual frames, only for image as whole.

While I understand that it makes sense in most cases to have some properties for all frames to be the same (like artist, copyright which you mentioned), as you also mentioned its not clear from the spec that this has to be the case. I think there can be cases where it makes sense to have different info's per frame, even though those might be admit ably more rare. I think we should keep it per frame as it is right now.

> So, useful profile properties can be placed in image metadata. Also they can be placed in frame metadata and user can access from first frame metadata.

I think having properties in image metadata and also in image frame metadata makes it more complicated. We then need to keep those in sync somehow.

Reply:
> I think having properties in image metadata and also in image frame metadata makes it more complicated. We then need to keep those in sync somehow.

@brianpopow  totally agree with you, this finally solves our dilemma